### ISSUE TITLE
kotlin: split on parameters in a function_declaration

### ISSUE THREAD

**Issue by Baehn** [Wansmer/treesj#55]
Thank you for the great plugin! :)

I have been able to configure language support for some kotlin language constructs but I have trouble with function declarations:
Its treesitter ast does not group function parameters in a common parent node like other languages do.

For example the function 
`fun test(a: String, b: String){
}
`

has the tree
<pre>
function_declaration [0, 0] - [1, 1]
  simple_identifier [0, 4] - [0, 8]
  parameter [0, 9] - [0, 18]
    simple_identifier [0, 9] - [0, 10]
    user_type [0, 12] - [0, 18]
      type_identifier [0, 12] - [0, 18]
  parameter [0, 20] - [0, 29]
    simple_identifier [0, 20] - [0, 21]
    user_type [0, 23] - [0, 29]
      type_identifier [0, 23] - [0, 29]
  function_body [0, 30] - [1, 1]
</pre>

Is it still possible to configure treesj for this case?

**Comment by Wansmer** [comment]
It is ts-structure of parameters has very difference of implementation in other languages.
But you can configure it with `omit`:
```lua
local kotlin = {
  function_declaration = u.set_preset_for_args({
    both = {
      omit = { '(', 'simple_identifier', 'function_body' },
    },
  }),
}
```

This solution has one problem – when you `join` it back, you will have extra spaces around brackets. At the moment it is not configurable with TreeSJ, but you can call your formatter after joining, if you don't need extra spaces.

From:
```kotlin
fun test(a:String, b: String){ } 
```
to:
```kotlin
fun test(
  a: String,
  b: String
){ }
```
and back only with extra spaces:
```kotlin
fun test( a: String, b: String ){ }
```

**Comment by Baehn** [comment]
thank you for your quick reply, unfortunately this does not work on functions with statements in its body:
<pre>
fun test(a:String, b: String) { 
    val var1 = 1
    val var2 = 2
}
</pre>
turns into 
<pre>fun test( a:String, b: String) { val var1 = 1 val var2 = 2 }</pre>

**Comment by Wansmer** [comment]
Yes, you are right. When I developed this plugin, I not meet similar structures, unfortunately.

Can you submit more examples with Kotlin, please? (before split/after split)
I will study examples and maybe could to add handling for similar structures in the future.

Btw, you can configure `function_body` and `statements` with:
```lua
{
  function_body = {
    target_nodes = { 'statements' },
  },
  statements = u.set_preset_for_non_bracket({
    join = {
      force_insert = ';',
    },
  }),
}
```

**Comment by Baehn** [comment]
sure, currently I have configured with 
<pre>
kotlin = {
        value_arguments = u.set_default_preset({
          both = {
            separator = ',',
          },
        }),
        collection_literal = u.set_default_preset({
          both = {
            separator = ',',
          },
        }),
        lambda_literal = u.set_default_preset({
          both = {
            separator = ';',
          },
        }),
        function_body = {
          target_nodes = { 'statements' },
        },
        statements = u.set_preset_for_non_bracket({
          join = {
            force_insert = ';',
          },
        }),
      }
</pre>

this works good for e.g.
<pre>
val collection_literal = [1, 2, 3]
val list = listOf(1, 2, 3)
call_expression("arg1", 2)
</pre>

expression
<pre>
   call_expression("arg1") {
       call1(1, 2)
       call2(1, 2)
   }
</pre>
transforms to 
<pre>
   call_expression("arg1") { call1(1, 2); call2(1, 2); }
</pre>
would be nice to get rid of the last `;`

expression
<pre>
   call_expression("arg1") {
       call1()
   }
</pre>
transforms to
<pre>
   call_expression("arg1") { call1(); }
</pre>
would be nice to get rid of the last `;` too

I can post more use cases if I encounter them during work

**Comment by Wansmer** [comment]
Thanks!

> would be nice to get rid of the last ;

You can configure this node with:
```lua
    join = {
      no_insert_if = {
        u.no_insert.if_penultimate,
      },
    },
```
(It is not documented yet, but works)

**Comment by Wansmer** [comment]
Implemented in #80
